routes/api/stream/index.js

```diff
@@ -30,29 +29,18 @@ router.get('/', (req, res, next) => {
     // Get the moderation setting from the settings.
     Setting.getModerationSetting()
   ])
-  .then(([asset, {moderation}]) => {
-    let comments;
-
-    if (moderation === 'post') {
-      comments = Comment.findAcceptedByAssetId(asset.id);
-    } else {
-
-      // Defaults to 'pre' moderation.
-      comments = Comment.findAcceptedAndNewByAssetId(asset.id);
+  .then(([asset, settings]) => {
+    // Get the sitewide moderation setting and return the appropriate comments
+    switch(settings.moderation){
+    case 'pre':
+      return Promise.all([Comment.findAcceptedByAssetId(asset.id), asset, settings]);
+    case 'post':
+      return Promise.all([Comment.findAcceptedAndNewByAssetId(asset.id), asset, settings]);
+    default:
+      return Promise.reject(new Error('Moderation setting not found.'));
     }
```

Comment:
The switch was replaced because the default described in the database is "pre". Because there can be only two states it made sense to not complain in the event of a missing setting. I'd prefer to retain this as the single if check as before and simply add the settings object to the returned `Promise.all` below.